share/transformers/mods.py

```diff
@@ -0,0 +1,499 @@
+    agent = tools.Delegate(MODSSimpleAgent, ctx)
+
+
+class MODSCreativeWork(Parser):
+    default_type = 'CreativeWork'
+    type_map = None
+    role_map = None
+
+    schema = tools.RunPython(
+        'get_schema',
+        tools.OneOf(
+            tools.RunPython(force_text, ctx['mods:genre']),
+            tools.Static(None)
+        )
+    )
+
+    title = tools.RunPython('join_title_info', ctx)
+
+    # Abstracts have the optional attribute "shareable". Don't bother checking for it, because
+    # abstracts that are not shareable should not have been shared with SHARE.
+    description = tools.Join(tools.RunPython(force_text, tools.Try(ctx['mods:abstract'])))
+
+    identifiers = tools.Map(
+        tools.Delegate(MODSWorkIdentifier),
+        tools.Unique(tools.Map(
+            tools.Try(tools.IRI(), exceptions=(ValueError, )),
+            tools.RunPython(
+                force_text,
+                tools.Filter(
+                    lambda obj: 'invalid' not in obj,
+                    tools.Concat(
+                        tools.Try(ctx['mods:identifier']),
+                        tools.Try(ctx.header['identifier'])
+                    )
+                )
+            )
+        ))
+    )
+
+    related_works = tools.Concat(
+        tools.Map(
+            tools.Delegate(MODSWorkRelation),
+            tools.Try(ctx['mods:relatedItem'])
+        )
+    )
+
+    related_agents = tools.Concat(
+        tools.Map(
+            tools.Delegate(MODSCreator),
+            tools.RunPython('filter_names', ctx, 'creator')
+        ),
+        tools.Map(
+            tools.Delegate(MODSFunder),
+            tools.RunPython('filter_names', ctx, 'funder')
+        ),
+        tools.Map(
+            tools.Delegate(MODSHost),
+            tools.RunPython('filter_names', ctx, 'host')
+        ),
+        tools.Map(
+            tools.Delegate(MODSPublisher),
+            tools.RunPython('filter_names', ctx, 'publisher')
+        ),
+        tools.Map(
+            tools.Delegate(MODSContributor),
+            tools.RunPython('filter_names', ctx, 'creator', 'funder', 'host', 'publisher', invert=True)
+        ),
+        tools.Map(
+            tools.Delegate(MODSSimplePublisher),
+            tools.Try(ctx['mods:originInfo']['mods:publisher']),
+        ),
+    )
+
+    rights = tools.RunPython(force_text, tools.Try(ctx['mods:accessCondition']))
+
+    language = tools.ParseLanguage(
+        tools.Try(ctx['mods:language']['mods:languageTerm']),
+    )
+
+    subjects = tools.Map(
+        tools.Delegate(MODSThroughSubjects),
+        tools.Subjects(
+            tools.Concat(
+                tools.Try(ctx['mods:subject']['mods:topic']),
+            )
+        )
+    )
+
+    tags = tools.Map(
+        tools.Delegate(MODSThroughTags),
+        tools.Concat(
+            tools.Map(
+                tools.RunPython('tokenize'),
+                tools.Map(
+                    tools.RunPython(force_text),
+                    tools.Concat(
```

Comment:
I don't think you need `Concat` when calling `Map`? It should implicitly call it for you.